tests/foreman/api/test_contentmanagement.py

```diff
@@ -125,6 +126,48 @@ def test_positive_sync_repos_with_lots_files(self):
         with self.assertNotRaises(TaskFailedError):
             repo.sync()
 
+    @tier2
+    def test_positive_kickstart_encoding_gzip(self):
+        """No encoding gzip errors on kickstart repositories
+        sync.
+
+        :id: dbdabc0e-583c-4186-981a-a02844f90412
+
+        :expectedresults: No encoding gzip errors present in /var/log/messages.
+
+        :CaseLevel: Integration
+
+        :steps:
+
+            1. Sync a kickstart repository.
+            2. After the repo is synced, change the download policy to
+                immediate.
+            3. Sync the repository again.
+            4. Assert that no errors related to encoding gzip are present in
+                /var/log/messages.
+
+        :CaseComponent: Pulp
+
+        :BZ: 1687801
+        """
+        org = entities.Organization().create()
+        product = entities.Product(organization=org).create()
+        repo = entities.Repository(
+            product=product,
+            url=CUSTOM_KICKSTART_REPO
+        ).create()
+        repo.sync()
+        repo = repo.read()
+        repo.download_policy = 'immediate'
+        repo = repo.update(['download_policy'])
+        repo.sync()
+        results = [
+            ssh.command('cat /var/log/messages | grep {}'.format(key))
+            for key in ('encoding', 'gzip', 'failed')
+        ]
+        self.assertTrue(all([result.return_code == 1 for result in results]))
+        self.assertTrue(all([len(result.stdout) == 0 for result in results]))
```

Comment:
```python
        self.assertTrue(all(result.return_code == 1 for result in results))
        self.assertTrue(all(len(result.stdout) == 0 for result in results))
```

`(x for x in items)` is generator expression - it gives one value at the time. It's good for functions like `all` or `any`, as they can exit earlier, on first value that determines entire result (i.e. on first False for `all`). Of course it doesn't make any difference for 3-element lists.

But cool thing about generator expressions is that you can omit parentheses if they are the only argument to the function. So you can write `all(x for x in items)` instead of `all([x for x in items])`.

Reply:
That is a great hint. Thanks